pkg/operator/controller/controller.go

```diff
@@ -160,6 +170,16 @@ func (r *ReconcileCDI) Reconcile(request reconcile.Request) (reconcile.Result, e
 
 	// mid delete
 	if cr.DeletionTimestamp != nil {
+		waitForDVs, err := r.waitForDVs(reqLogger, cr)
+		if err != nil {
+			return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Second}, err
+		}
+
+		if waitForDVs {
+			reqLogger.Info("Not deleting as long as DVs exist")
+			return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
```

Comment:
Just so I understand, we are rechecking every 5 seconds if all the dvs are gone, once the CR is marked for deletion?

Reply:
Yes